application/src/main/java/uk/nhs/hee/tis/revalidation/validator/TraineeRecommendationRecordDTOValidator.java

```diff
@@ -73,4 +84,14 @@ public void validate(Object target, Errors errors) {
       }
     }
   }
+
+  private LocalDate getDoctorGmcSubmissionDueDate(TraineeRecommendationRecordDto recordDto) {
+    final var doctorsForDB = doctorsForDBRepository.findById(recordDto.getGmcNumber());
+    if (doctorsForDB.isEmpty()) {
+      throw new RecommendationException(
+          format(DOCTOR_NOT_FOUND_MESSAGE, recordDto.getGmcNumber()));
```

Comment:
The error throwing doesn't look sensible to me. 
I'm wondering why don't throw the error to the caller method, and have a `errors.reject` as others do.

Reply:
Need some mobbing here. Not sure how to do it.

Reply:
The couple that jump to mind are:
1. Merge this into the method.
2. Pass `errors` as a parameter
3. Have a null safe comparison in the calling method if there's no significant difference between a) no doctor & b) null submission date*
4. Return the persisted doctor (as there might be other checks to do on the persisted doctor)*

*These might be ok but don't jump out as natural ways to do it.